// @100mslive/server-sdk/dist/esm/api_wrappers/LiveStreamWrapper.js
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
import { QueryObjectIterator } from "../utils/QueryObjectIterator";
/**
 * The wrapper class that implements all
 * {@link https://www.100ms.live/docs/server-side/v2/api-reference/live-streams/overview Live Stream API} calls.
 */
export default class LiveStreamWrapper {
    constructor(apiService) {
        this.apiService = apiService;
        this.basePath = "/live-streams";
    }
    /**
     * Get a list of live stream objects that satisfy the `filter` params. A
     * `HMS.LiveStream.Object` iterable is returned that can be iterated with a `for await` loop.
     * @param filters Live stream filters like room ID and status
     * @returns a `HMS.QueryObjectIterator<HMS.LiveStream.Object>` object
     */
    list(filters) {
        const queryObjectIterable = new QueryObjectIterator((queryParams) => this.apiService.get(this.basePath, queryParams), filters);
        return queryObjectIterable;
    }
    /**
     * Get the details of a live stream by stream id.
     * @param streamId Stream ID
     * @returns a `HMS.LiveStream.Object` object
     */
    retrieve(streamId) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.apiService.get(`${this.basePath}/${streamId}`);
        });
    }
    /**
     * Start a new live stream for a room.
     * @param roomId Room ID
     * @param params Params to start a live stream
     * @returns a `HMS.LiveStream.Object` object
     */
    start(roomId, params) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.apiService.post(`${this.basePath}/room/${roomId}/start`, params);
        });
    }
    /**
     * Send a timed metadata to an existing live stream. The `payload` in params
     * will be set as the metadata in the live stream, after `duration` seconds.
     * @param streamId Stream ID
     * @param params Timed Metadata params
     * @returns a `HMS.LiveStream.Object` object
     */
    sendTimedMetadata(streamId, params) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.apiService.post(`${this.basePath}/${streamId}/timed-metadata`, params);
        });
    }
    /**
     * Pause recording for a specific live stream.
     * @param streamId Stream ID
     * @returns a `HMS.LiveStream.Object` object
     */
    pauseRecording(streamId) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.apiService.post(`${this.basePath}/${streamId}/pause-recording`);
        });
    }
    /**
     * Resume recording for a specific live stream.
     * @param streamId Stream ID
     * @returns a `HMS.LiveStream.Object` object
     */
    resumeRecording(streamId) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.apiService.post(`${this.basePath}/${streamId}/resume-recording`);
        });
    }
    /**
     * Stop live stream by stream id.
     * @param streamId Stream ID
     * @returns a `HMS.LiveStream.Object` object
     */
    stop(streamId) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.apiService.post(`${this.basePath}/${streamId}/stop`);
        });
    }
    /**
     * Stop all live streams for a room.
     * @param roomId Room ID
     * @returns a `HMS.LiveStream.Object[]` object
     */
    stopAll(roomId) {
        var _a;
        return __awaiter(this, void 0, void 0, function* () {
            const results = yield this.apiService.post(`${this.basePath}/room/${roomId}/stop`);
            return (_a = results.data) !== null && _a !== void 0 ? _a : [];
        });
    }
}
