// @100mslive/server-sdk/dist/cjs/types/api_types/policy.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.SimulcastVideoLayerRid = exports.VideoCodec = exports.AudioCodec = exports.AllowedTracks = void 0;
var AllowedTracks;
(function (AllowedTracks) {
    AllowedTracks["Audio"] = "audio";
    AllowedTracks["Screen"] = "screen";
    AllowedTracks["Video"] = "video";
    AllowedTracks["Simulcast"] = "simulcast";
})(AllowedTracks = exports.AllowedTracks || (exports.AllowedTracks = {}));
var AudioCodec;
(function (AudioCodec) {
    AudioCodec["Opus"] = "opus";
})(AudioCodec = exports.AudioCodec || (exports.AudioCodec = {}));
var VideoCodec;
(function (VideoCodec) {
    VideoCodec["VP8"] = "vp8";
})(VideoCodec = exports.VideoCodec || (exports.VideoCodec = {}));
var SimulcastVideoLayerRid;
(function (SimulcastVideoLayerRid) {
    SimulcastVideoLayerRid["F"] = "f";
    SimulcastVideoLayerRid["H"] = "h";
    SimulcastVideoLayerRid["Q"] = "q";
})(SimulcastVideoLayerRid = exports.SimulcastVideoLayerRid || (exports.SimulcastVideoLayerRid = {}));
