// @100mslive/server-sdk/dist/cjs/types/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    var desc = Object.getOwnPropertyDescriptor(m, k);
    if (!desc || ("get" in desc ? !m.__esModule : desc.writable || desc.configurable)) {
      desc = { enumerable: true, get: function() { return m[k]; } };
    }
    Object.defineProperty(o, k2, desc);
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __exportStar = (this && this.__exportStar) || function(m, exports) {
    for (var p in m) if (p !== "default" && !Object.prototype.hasOwnProperty.call(exports, p)) __createBinding(exports, m, p);
};
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.Session = exports.Room = exports.RoomCode = exports.Recording = exports.RecordingAsset = exports.Policy = exports.LiveStream = exports.ExternalStream = exports.Analytics = exports.ActiveRoom = exports.Common = void 0;
__exportStar(require("./auth"), exports);
__exportStar(require("./exception"), exports);
__exportStar(require("./pagination"), exports);
// API types
const Common = __importStar(require("./api_types/common"));
exports.Common = Common;
const ActiveRoom = __importStar(require("./api_types/activeRoom"));
exports.ActiveRoom = ActiveRoom;
const Analytics = __importStar(require("./api_types/analytics"));
exports.Analytics = Analytics;
const ExternalStream = __importStar(require("./api_types/externalStream"));
exports.ExternalStream = ExternalStream;
const LiveStream = __importStar(require("./api_types/liveStream"));
exports.LiveStream = LiveStream;
const Policy = __importStar(require("./api_types/policy"));
exports.Policy = Policy;
const RecordingAsset = __importStar(require("./api_types/recordingAsset"));
exports.RecordingAsset = RecordingAsset;
const Recording = __importStar(require("./api_types/recording"));
exports.Recording = Recording;
const RoomCode = __importStar(require("./api_types/roomCode"));
exports.RoomCode = RoomCode;
const Room = __importStar(require("./api_types/room"));
exports.Room = Room;
const Session = __importStar(require("./api_types/session"));
exports.Session = Session;
