// @100mslive/server-sdk/dist/cjs/api_wrappers/ExternalStreamWrapper.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
const QueryObjectIterator_1 = require("../utils/QueryObjectIterator");
/**
 * The wrapper class that implements all
 * {@link https://www.100ms.live/docs/server-side/v2/api-reference/external-streams/overview External Stream API} calls.
 */
class ExternalStreamWrapper {
    constructor(apiService) {
        this.apiService = apiService;
        this.basePath = "/external-streams";
    }
    /**
     * Get a list of external stream objects that satisfy the `filter` params. A
     * `HMS.ExternalStream.Object` iterable is returned that can be iterated with a `for await` loop.
     * @param filters External stream filters like room ID and status
     * @returns a `HMS.QueryObjectIterator<HMS.ExternalStream.Object>` object
     */
    list(filters) {
        const queryObjectIterable = new QueryObjectIterator_1.QueryObjectIterator((queryParams) => this.apiService.get(this.basePath, queryParams), filters);
        return queryObjectIterable;
    }
    /**
     * Get the details of an external stream by stream id.
     * @param streamId Stream ID
     * @returns a `HMS.ExternalStream.Object` object
     */
    retrieve(streamId) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.apiService.get(`${this.basePath}/${streamId}`);
        });
    }
    /**
     * Start a new external stream in a room.
     * @param roomId Room ID
     * @param params Params to start an external stream
     * @returns a `HMS.ExternalStream.Object` object
     */
    start(roomId, params) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.apiService.post(`${this.basePath}/room/${roomId}/start`, params);
        });
    }
    /**
     * Stop an external stream by stream id.
     * @param streamId Stream ID
     * @returns a `HMS.ExternalStream.Object` object
     */
    stop(streamId) {
        return __awaiter(this, void 0, void 0, function* () {
            return this.apiService.post(`${this.basePath}/${streamId}/stop`);
        });
    }
    /**
     * Stop all external streams for a room.
     * @param roomId Room ID
     * @returns a `HMS.ExternalStream.Object[]` object
     */
    stopAll(roomId) {
        var _a;
        return __awaiter(this, void 0, void 0, function* () {
            const results = yield this.apiService.post(`${this.basePath}/room/${roomId}/stop`);
            return (_a = results.data) !== null && _a !== void 0 ? _a : [];
        });
    }
}
exports.default = ExternalStreamWrapper;
